Kuzzle/API/Controllers/BulkController.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+
+namespace KuzzleSdk.API.Controllers {
+  /// <summary>
+  /// Implements the "bulk" Kuzzle API controller
+  /// </summary>
+  public sealed class BulkController : BaseController {
+    internal BulkController(IKuzzleApi api) : base(api) { }
+
+    /// <summary>
+    /// Creates, updates or deletes large amounts of documents as fast as possible.
+    /// </summary>
+    public async Task<JObject> ImportAsync(
+        string index,
+        string collection,
+        JArray bulkData
+    ) {
+      Response response = await api.QueryAsync(new JObject {
+        {"index", index},
+        {"collection", collection},
+        {"controller", "bulk"},
+        {"action", "import"},
+        {"body", new JObject {
+            {"bulkData", bulkData}
+          }
+        }
+      });
+      return (JObject)response.Result;
+    }
+
+    /// <summary>
+    /// Create or replace multiple documents directly into the storage engine.
+    /// </summary>
+    public async Task<JObject> mWriteAsync(
+        string index,
+        string collection,
+        JArray documents,
+        string refresh = null,
+        bool? notify = null
+      ) {
+
+      Response response = await api.QueryAsync(new JObject {
+        {"index", index},
+        {"collection", collection},
+        {"controller", "bulk"},
+        {"action", "mWrite"},
+        {"notify", notify},
+        {"refresh", refresh},
+        {"body", new JObject {
+            {"documents", documents}
+          }
+        }
+      });
+
+      return (JObject)response.Result;
+    }
+
+    /// <summary>
+    /// Create or replace a document directly into the storage engine.
+    /// </summary>
+    public async Task<JObject> WriteAsync(
+        string index,
+        string collection,
+        string documentContent,
+        string documentId = null,
+        string refresh = null,
```

Comment:
All SDKs replaces this API option with a boolean `WaitForRefresh` one. Under the hood, if set to true, the request is sent with a `refresh: 'wait_for'` value set in the JSON payload.
(check the document controller for instance)

In the next major version of Kuzzle, the API option `refresh: 'wait_for'` will probably be deprecated in favor of a `waitForRefresh: <boolean>` option.


(I won't report this issue more than once but it must be fixed everywhere obviously)

Reply:
This should be resolved now, I changed everything related to the refresh option.
now there is a QueryUtils class with static methods [HandleRefreshOption, HandleNotifyOption]
that do the job and I changed every Controller methods that have a refresh parameter to use QueryUtils.HandleRefreshOption instead and avoid copy paste.
Same thing for every unit tests replace `if (refresh) expected.Add("refresh", "wait_for");` to use `QueryUtils.HandleRefreshOption(expected, refresh)` instead, this would be easier if the actual API option `refresh: "wait_for"` needs to be deprecated.

Reply:
I agree with all these changes but the last one: unit tests should add the property themselves without resorting to HandleRefreshOption. That way, we can make sure that JSON requests are properly built, testing the new Handlexxx methods along the way.

Reply:
This has been changed, now there is a QueryUtils inside Kuzzle.Tests